src/Hipace.cpp

```diff
@@ -817,6 +820,12 @@ Hipace::ExplicitMGSolveBxBy (const int lev, const int which_slice)
         m_hpmg[lev]->solve1(BxBy[0], SySx[0], Mult[0], m_MG_tolerance_rel, m_MG_tolerance_abs,
                             max_iters, m_MG_verbose);
     }
+
+    // interpolate Bx and By to lev from lev-1 in the ghost cells
+    m_fields.LevelUpBoundary(m_3D_geom, lev, which_slice, "Bx",
+        m_fields.m_slices_nguards, m_fields.m_poisson_nguards);
+    m_fields.LevelUpBoundary(m_3D_geom, lev, which_slice, "By",
+        m_fields.m_slices_nguards, m_fields.m_poisson_nguards);
```

Comment:
Also on this one?

Reply:
Because the nodal MG solver doesn't read in the ghost cells anymore, the level >=1 boundary condition has to come from SetBoundaryCondition instead of LevelUpBoundary.